### PR TITLE
Org level autosubmit configuration file

### PR DESCRIPTION
**Pull request by ricardoamador**
In order to simplify Configurable Autosubmit a change will be made to the configuration model to allow an org to specify a global configuration in the org/.github repository.

Setting allow_config_override will allow changes to the settings for individual repositories should they need to modify this global configuration.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: autosubmit/autosubmit.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-default_branch: main
+allow_config_override: false
```
Comment (keyonghan) [inline autosubmit/autosubmit.yml:5]: Do we not want to allow override? Based on description, sounds we allow sub repo to override.

Comment (ricardoamador) [reply autosubmit/autosubmit.yml:5]: We can, but for our (Flutter's) supported repositories it currently does not matter as the settings across our repositories are the same. Perhaps in the future I will need to change this.

Comment (ricardoamador) [reply autosubmit/autosubmit.yml:5]: I can enable it if you feel it is needed? It is not a big deal in our repos unless someone allows a config to be committed.

Comment (keyonghan) [reply autosubmit/autosubmit.yml:5]: My feeling is this config provides flexibility for different repos to config separately. Disallowing it sounds against it. As you mentioned, all configs are the same now. You call to add or not:-)

**Review by keyonghan — DISMISSED** [review]
LGTM

File: autosubmit/autosubmit.yml (hunk 1 above)
Comment (ricardoamador) [reply autosubmit/autosubmit.yml:5]: Yeah exactly! It is for allowing the repos to configure separately. We are currently protected unless someone allows a config change in the child repo. I think I am going to enable it temporarily and then once we are servicing the dart team I will disable it.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**